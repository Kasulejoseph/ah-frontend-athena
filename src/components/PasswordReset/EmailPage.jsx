import React from 'react'
import "./PasswordReset.scss";

export const EmailPage = () => (
  <div className="w-100 flex flex-center flex-column align-items-center justify-content-center">
    <h2 className="blue-text">An email has been sent</h2>
    <h4 className="">
      Please check your email and follow the 
      instructions to reset your passsword
    </h4>
  </div>
  );


export default EmailPage;
