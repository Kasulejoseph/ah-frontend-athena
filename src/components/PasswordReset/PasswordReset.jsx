import React from 'react';
import PropTypes from 'prop-types';
import { MDBInput, MDBBtn, MDBContainer, MDBCol, MDBRow, MDBCard, MDBCardBody } from "mdbreact";

const PasswordResetView = (props) => {
  const {onSubmit,onChange,email,errors}=props
  return (
    <MDBContainer className="w-75 mt-5 pt-5 mt-5 mb-5">
      <MDBRow className="flex flex-center">
        <MDBCol md="6">
          <MDBCard>
            <div className="header pt-3 blue lighten-2">
              <MDBRow className="d-flex justify-content-start">
                <h4 className="text-white mt-3 mb-4 pb-1 mx-5">
                  Forgot Password?
                </h4>
              </MDBRow>
            </div>
            <MDBCardBody>
              <form onSubmit={onSubmit}>
                <div className="grey-text">
                  <MDBInput
                    name="email"
                    label="Type your email"
                    icon="envelope"
                    group
                    type="email"
                    validate
                    error="wrong"
                    success="right"
                    onChange={onChange}
                    value={email}
                    required
                  />
                  <span className=" red-text is-invalid">{errors.email}</span>
                  <MDBBtn className="form-control" color="info" fontWeight="600" type='submit'>Submit</MDBBtn>
                </div>
              </form>
            </MDBCardBody>
          </MDBCard>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  )
}

PasswordResetView.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  email: PropTypes.shape({}),
  errors: PropTypes.shape({})
}

PasswordResetView.defaultProps = {
  email: {},
  errors: {}
}

export default PasswordResetView;
