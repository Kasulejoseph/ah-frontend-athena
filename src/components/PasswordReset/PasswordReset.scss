.form-control {
  margin-bottom: 0;
}

.red-text {
  font-size: 90%;
  margin-bottom: 50px;
}

.btn {
  margin-bottom: 20px;
}

.full-height {
  height: 100vw !important;
}
