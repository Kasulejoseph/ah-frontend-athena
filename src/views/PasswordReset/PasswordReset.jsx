import React, { Component } from 'react'
import {connect} from 'react-redux';
import PropTypes from "prop-types";
import PasswordResetView from '../../components/PasswordReset/PasswordReset';
import { PasswordResetAction } from '../../actions/PasswordResetAction/PasswordResetAction';

export class PasswordReset extends Component {

  constructor(props){
    super(props);
    this.state = {
      email: '',
      errors: ''
    }
  }

  componentWillReceiveProps(nextProps){
    if(Object.keys(nextProps.message).length !== 0){
      window.location.href='/email-has-been-sent'
      
    }else if(Object.keys(nextProps.errors).length !== 0){
        this.setState({errors:nextProps.errors})
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const {email}=this.state
    const {PasswordResetAction}=this.props
    const data = {
      email: email
    }
    PasswordResetAction(data)
  };

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  };

  render() {
    const {email, errors}=this.state
    return (
      <PasswordResetView
        onSubmit={this.handleSubmit}
        onChange={this.handleChange} 
        email={email}
        errors={errors}
      />
    );
  }
}

PasswordReset.propTypes = {
  message: PropTypes.shape({}),
  errors: PropTypes.shape({}),
  PasswordResetAction: PropTypes.func.isRequired
}

PasswordReset.defaultProps = {
  message: {},
  errors: {}
}

export const mapStateToProps = (state)=>{
  
  return{
    errors: state.passwordResetConfirm.errors,
    message: state.passwordResetConfirm.message
  }
}

export default connect(mapStateToProps, {
  PasswordResetAction
})(PasswordReset);
