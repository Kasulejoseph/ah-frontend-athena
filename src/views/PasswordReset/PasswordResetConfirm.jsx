import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import PropTypes from "prop-types";
import PasswordResetConfirmView from '../../components/PasswordReset/PasswordResetConfirm';
import { PasswordResetConfirmAction } from '../../actions/PasswordResetAction/PasswordResetAction';

export class PasswordResetConfirm extends Component {

  constructor(){
    super();
    this.state = {
      password: '',
      confirmPassword: '',
      errors: '',
    }
  }

  componentWillReceiveProps(nextProps){
    if(Object.keys(nextProps.message).length !== 0){
      const {message}=nextProps
      toast.success(message.detail.toString());
      window.location.href='/login';
    }else if(Object.keys(nextProps.errors).length !== 0){
        this.setState({errors:nextProps.errors})
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const {password, confirmPassword}=this.state
    const {PasswordResetConfirmAction}=this.props
    const data = {
      password: password,
      confirm_password: confirmPassword
    }
    const pathName = window.location.pathname.split('/')
    PasswordResetConfirmAction(data, pathName)
  };

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  };

  render() {
    const {errors, password, confirmPassword}=this.state
    return (
      <PasswordResetConfirmView 
        errors={errors}
        onSubmit={this.handleSubmit}
        onChange={this.handleChange} 
        newPassword={password}
        confirmPassword={confirmPassword}
      />
    )
  }
}

PasswordResetConfirm.propTypes = {
  message: PropTypes.shape({}),
  errors: PropTypes.shape({}),
  PasswordResetConfirmAction: PropTypes.func.isRequired
}

PasswordResetConfirm.defaultProps = {
  message: {},
  errors: {},
}

export const mapStateToProps = (state) => {
  
  return{
    errors: state.passwordResetConfirm.errors,
    message: state.passwordResetConfirm.message
  }
}

export default connect(mapStateToProps, {
  PasswordResetConfirmAction
})(PasswordResetConfirm);
