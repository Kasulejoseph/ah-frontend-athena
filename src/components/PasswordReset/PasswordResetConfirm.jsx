import React from 'react';
import PropTypes from 'prop-types';
import { MDBInput, MDBBtn, MDBContainer, MDBCol, MDBRow, MDBCard, MDBCardBody  } from "mdbreact";
import './PasswordReset.scss';

const PasswordResetConfirmView = (props) => {
  const {onSubmit,onChange,newPassword,confirmPassword,errors}=props
  return (
    <MDBContainer className="w-75 mt-5 pt-5 mt-5 mb-5">
      <MDBRow className="flex flex-center">
        <MDBCol md="6">
          <MDBCard>
            <div className="header pt-3 blue lighten-2">
              <MDBRow className="d-flex justify-content-start">
                <h4 className="text-white mt-3 mb-4 pb-1 mx-5">
                  Reset Password
                </h4>
              </MDBRow>
            </div>
            <MDBCardBody>
              <form onSubmit={onSubmit}>
                <div className="grey-text">
                  <MDBInput
                    name="password"
                    label="New Password"
                    icon="lock"
                    group
                    type="password"
                    validate
                    error="wrong"
                    success="right"
                    onChange={onChange}
                    value={newPassword}
                    required
                  />
                  <MDBInput
                    name="confirmPassword"
                    label="Confirm Password"
                    icon="lock"
                    group
                    type="password"
                    validate
                    onChange={onChange}
                    value={confirmPassword}
                    required
                  />
                  <span className=" red-text is-invalid">{errors.body}</span>
                  <span className=" red-text is-invalid">{errors.password}</span>
                </div>
                <div className="text-center">
                  <MDBBtn className="form-control" color="info" fontWeight="600" type="submit">Submit</MDBBtn>
                </div>
              </form>
            </MDBCardBody>
          </MDBCard>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  )
}

PasswordResetConfirmView.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  errors: PropTypes.shape({}),
  newPassword: PropTypes.shape({}),
  confirmPassword: PropTypes.shape({})
}

PasswordResetConfirmView.defaultProps = {
  newPassword: {},
  confirmPassword: {},
  errors: {}
}

export default PasswordResetConfirmView;
