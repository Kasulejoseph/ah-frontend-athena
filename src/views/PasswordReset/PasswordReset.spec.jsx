import React from 'react';
import { shallow } from 'enzyme';
import { PasswordReset, mapStateToProps } from './PasswordReset';
import { PasswordResetConfirm } from './PasswordResetConfirm';
import * as obj from './PasswordResetConfirm';

let props2 = {
  onSubmit: jest.fn(),
  onChange: jest.fn(),
  PasswordResetConfirmAction: jest.fn(),
  newPassword: {},
  confirmPassword: {},
}
let props = {
  onSubmit: jest.fn(),
  onChange: jest.fn(),
  PasswordResetAction: jest.fn(),
  email: {},
  errors: {}
}

const initialState = {
  passwordResetConfirm:{
  email: '',
  errors: 'wrong email'}
}
const initialState2 = {
  passwordResetConfirm:{
  message: '',
  errors: 'password not equal'}
}

let wrapper;
let otherWrapper;

describe('test password reset component', () => {
  beforeEach(() => {
    wrapper = shallow(<PasswordReset {...props} />)
  });
  it('compare snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call handle change', () => {
    wrapper.instance().handleChange({target: {name:'email',value:'josh@andela.com'}})
    expect(wrapper.state('email')).toEqual('josh@andela.com')
  })

  it('should handle submit', () => {
    wrapper.instance().handleSubmit({preventDefault() {}})
    expect(wrapper.state('email')).toEqual('')
  })
  it('will recieve next props success', ()=> {
    wrapper.setProps({message:{email: 'josh@gmail.com'}})
    expect(wrapper.state('email')).toEqual('')
   })
   it('will not recieve next props errors', ()=> {
    wrapper.setProps({message:{errors:"error message"}})
    expect(wrapper.state('errors')).toEqual('')
   })
  it('test mapStateToProps', () => {
    expect(mapStateToProps(initialState).errors).toEqual('wrong email')
  })
}); 

describe('test password reset confirm component', () => {
  beforeEach(() => {
    otherWrapper = shallow(<PasswordResetConfirm {...props2} />)
  });
  it('compare snapshot', () => {
    expect(otherWrapper).toMatchSnapshot()
  });
  it('should call handle change', () => {
    otherWrapper.instance().handleChange({target: {name:'password',value:'qwerty123'}})
    expect(otherWrapper.state('password')).toEqual('qwerty123')
  })
  it('should handle submit', () => {
    otherWrapper.instance().handleSubmit({preventDefault() {}})
    expect(otherWrapper.state()).toEqual({"confirmPassword": "", "errors": "", "password": ""})
  })
  it('will receieve next props', ()=> {
    otherWrapper.setProps({message:{detail:'josh@me.com'}})
    expect(otherWrapper.state().password).toEqual('')
   })
  it('test mapStateToProps', () => {
    expect(obj.mapStateToProps(initialState2).errors).toEqual("password not equal")
  })
}); 
