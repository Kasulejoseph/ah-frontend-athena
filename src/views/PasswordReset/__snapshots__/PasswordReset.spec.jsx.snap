// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test password reset component compare snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordReset
    PasswordResetAction={[MockFunction]}
    email={Object {}}
    errors={Object {}}
    message={Object {}}
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "email": "",
      "errors": "",
      "onChange": [Function],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "email": "",
        "errors": "",
        "onChange": [Function],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test password reset confirm component compare snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordResetConfirm
    PasswordResetConfirmAction={[MockFunction]}
    confirmPassword={Object {}}
    errors={Object {}}
    message={Object {}}
    newPassword={Object {}}
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "confirmPassword": "",
      "errors": "",
      "newPassword": "",
      "onChange": [Function],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "confirmPassword": "",
        "errors": "",
        "newPassword": "",
        "onChange": [Function],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
